### ISSUE TITLE
<style> is missing it's Part

### ISSUE THREAD

**Issue by KenBrowne** [PolymerLabs/lit-html#131]
Hello,
First time issue reporter, long time user (well... for about 1 month now)...
I installed version 0.7.1 and it has broken our current implementation upon rendering the `<style>` tag.

I have written a short CodePen to demonstrate:
[CodePen](https://codepen.io/kenbrowne/pen/xXjQoZ)

![screen shot 2017-10-10 at 17 23 17](https://user-images.githubusercontent.com/859524/31397947-4ecc4010-addf-11e7-99de-88e9eb6d2164.png)

As you can see from the above image it leaves the `textMarker` between the `<style>` tags and inserts the `Part` earmarked for the `<style>` tag inside the `<h1>` tags.
 
This is working as expected in 0.6.0.

**Comment by daKmoR** [comment]
dynamically creating one style tag with changing content is not optimal for browser optimization as far as I know... better have multiple style tags..
lit-html is also strong when it comes to merging it own templates... 

You could use it something like this https://codepen.io/daKmoR/pen/gGzqGY
```js
  var style = html`<style>h1 { color:red; }</style>`;
  render(html`${style}<h1>I should be red :(</h1> `, 
```

as far as I see it lit-html is for tag content replacement not for string concatenation. If you want that you can do it an then afterwards use unsafeHTML.
e.g. https://codepen.io/daKmoR/pen/aLGMoP
```js
  var style = 'h1{color:red;}';
  var rawStyle = `<style>${style}</style>`;
  var styleInsert = html`${unsafeHTML(rawStyle)}`;
  var text = 'I should be red :(';
  render(html`${styleInsert}<h1>${text}</h1>`, document.body);
```

PS: not sure if something like that should be supported by `lit-html`... I guess @justinfagnani has to verify

**Comment by KenBrowne** [comment]
Thanks @daKmoR,
I hadn't considered using unsafeHTML as an alternative due to the cross-site-scripting vulnerabilities. It certainly does work but It's unsafe to use. 

I get your point about String concatenation so here is a further insight as to how we are using lit-html:
We allow the developer to specify their Web Component style in a separate file giving the structure as:

```
//myWebComponent.css
h1 {
   color: red;
}
```   

```
//MyWebComponent.js
import style from './myWebComponent.css';
...
const content = html`<h1>This should be red :(</h1>`;
...

render(html`<style>${style}</style>${content}`, shadowRoot);
```

This is a simplified view of what we're trying to do.  Every Web Component has the ability to set their own CSS which will be housed within the shadowRoot of the Web Component.

**Comment by justinfagnani** [comment]
Thanks for the report @KenBrowne!

I unintentionally broke style bindings when I switched to using comments for placeholders in text node positions. I didn't know that HTML comments in style tags didn't create comment nodes! It was a simple fix in #135, and now we have tests covering style tags :)

**Comment by KenBrowne** [comment]
Thanks a million @justinfagnani :)
